ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -0,0 +1,132 @@
+        queuedIngestClient = new QueuedIngestClient(resourceManager, storageClient);
+        streamingIngestClient = new StreamingIngestClient(streamingClient);
+    }
+
+    @Override
+    public IngestionResult ingestFromFile(FileSourceInfo fileSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(fileSourceInfo, "fileSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        fileSourceInfo.validate();
+        ingestionProperties.validate();
+        try {
+            StreamSourceInfo streamSourceInfo = IngestionUtils.fileToStream(fileSourceInfo);
+            return ingestFromStream(streamSourceInfo, ingestionProperties);
+        } catch (FileNotFoundException e) {
+            log.error("File not found when ingesting a file.", e);
+            throw new IngestionClientException("IO exception - check file path.", e);
+        }
+
+    }
+
+    @Override
+    public IngestionResult ingestFromBlob(BlobSourceInfo blobSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(blobSourceInfo, "blobSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        blobSourceInfo.validate();
+        ingestionProperties.validate();
+
+        //if it's a blob we ingest using the queued client
+        return queuedIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
+    }
+
+    @Override
+    public IngestionResult ingestFromResultSet(ResultSetSourceInfo resultSetSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(resultSetSourceInfo, "resultSetSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        resultSetSourceInfo.validate();
+        ingestionProperties.validate();
+        try {
+            StreamSourceInfo streamSourceInfo = IngestionUtils.resultSetToStream(resultSetSourceInfo);
+            return ingestFromStream(streamSourceInfo, ingestionProperties);
+        } catch (IOException ex) {
+            String msg = "Failed to read from ResultSet.";
+            log.error(msg, ex);
+            throw new IngestionClientException(msg, ex);
+        }
+    }
+
+    @Override
+    public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, IngestionProperties ingestionProperties) throws IngestionClientException, IngestionServiceException {
+        Ensure.argIsNotNull(streamSourceInfo, "streamSourceInfo");
+        Ensure.argIsNotNull(ingestionProperties, "ingestionProperties");
+
+        streamSourceInfo.validate();
+        ingestionProperties.validate();
+
+        if (streamSourceInfo.isLeaveOpen()) {
+            throw new UnsupportedOperationException("Stream can't be Leave Open in ManagedStreamingIngestClient");
```

Comment:
Instead of "Leave Open", I suggest to write the meaning (alternatively: "LeaveOpen can't be true in ManagedStreamingIngestClient").

Reply:
I'm going with the second suggestion since it indicates the field name directly, which is clearer to me.